modules/instance_template/variables.tf

```
@@ -0,0 +1,123 @@
+ * limitations under the License.
+ */
+
+variable "name_prefix" {
+  description = "Name prefix for the instance template"
+  default     = "default-instance-template"
+}
+
+variable "machine_type" {
+  description = "Machine type to create, e.g. n1-standard-1"
+  default     = "n1-standard-1"
+}
+
+variable "can_ip_forward" {
+  description = "Enable IP forwarding, for NAT instances for example"
+  default     = "false"
+}
+
+variable "tags" {
+  type        = "list"
+  description = "Network tags, provided as a list"
+  default     = []
+}
+
+variable "labels" {
+  type        = "map"
+  description = "Labels, provided as a map"
+  default     = {}
+}
+
+#######
+# disk
+#######
+variable "source_image" {
+  description = "Source disk image. If neither source_image nor source_image_family is specified, defaults to the latest public CentOS image."
+  default     = ""
+}
+
+variable "source_image_family" {
+  description = "Source image family. If neither source_image nor source_image_family is specified, defaults to the latest public CentOS image."
+  default     = ""
+}
+
+variable "source_image_project" {
+  description = "Project where the source image comes from"
+  default     = ""
+}
+
+variable "disk_size_gb" {
+  description = "Boot disk size in GB"
+  default     = "100"
+}
+
+variable "disk_type" {
+  description = "Boot disk type, can be either pd-ssd, local-ssd, or pd-standard"
+  default     = "pd-standard"
+}
+
+variable "auto_delete" {
+  description = "Whether or not the boot disk should be auto-deleted"
+  default     = "true"
+}
+
+variable "additional_disks" {
+  description = "List of maps of additional disks. See https://www.terraform.io/docs/providers/google/r/compute_instance_template.html#disk_name"
+  type        = "list"
+  default     = []
+}
+
+####################
+# network_interface
+####################
+variable "network" {
+  description = "The name or self_link of the network to attach this interface to. Use network attribute for Legacy or Auto subnetted networks and subnetwork for custom subnetted networks."
+  default     = ""
+}
+
+variable "subnetwork" {
+  description = "The name of the subnetwork to attach this interface to. The subnetwork must exist in the same region this instance will be created in. Either network or subnetwork must be provided."
+  default     = ""
+}
+
+variable "subnetwork_project" {
+  description = "The ID of the project in which the subnetwork belongs. If it is not provided, the provider project is used."
+  default     = ""
+}
+
+###########
+# metadata
+###########
+
+variable "startup_script" {
+  description = "User startup script to run when instances spin up"
+  default     = ""
+}
+
+variable "metadata" {
+  type        = "map"
+  description = "Metadata, provided as a map"
+  default     = {}
+}
+
+##################
+# service_account
+##################
+
+variable "service_account" {
+  type        = "map"
+  description = "Service account email address and scopes"
```

This description should be more informative, and propagated to other modules as necessary.
```suggestion
  description = "Service account to attach to the instance. See https://www.terraform.io/docs/providers/google/r/compute_instance_template.html#service_account."
```